### PR TITLE
#818: unify RFC 3339 timestamp format strubg

### LINKED ISSUE #818: Unify conversion of datetime <--> timestamp strings
Use a shared formatting string for the "RFC 3339 w/ microseconds" format, and centralize conversion of datetimes to strings and back (w/ `pytz.utc` zone).

See:

- https://github.com/GoogleCloudPlatform/gcloud-python/pull/816#discussion_r28172973
- https://github.com/GoogleCloudPlatform/gcloud-python/pull/816#discussion_r28173336
- https://github.com/GoogleCloudPlatform/gcloud-python/pull/816#discussion_r28180427

### PR DESCRIPTION
**Pull request by tseaver**
Use consistently across `storage` and `pubsub`.

Fixes #818.

~~Uses #816 as a diffbase.~~

### CODE REVIEW

**Comment by tseaver** [comment]
Rebased after merge of #816 to master.

**Comment by dhermes** [comment]
LGTM. Thanks for doing this!

**Comment by dhermes** [comment]
Actually wait, don't we want to put a timezone on the stamps in storage?

File: gcloud/storage/blob.py [hunk 1]
```diff
@@ -749,7 +749,7 @@ def time_deleted(self):
         """
         value = self._properties.get('timeDeleted')
         if value is not None:
-            return datetime.datetime.strptime(value, _GOOGLE_TIMESTAMP_FORMAT)
+            return datetime.datetime.strptime(value, _RFC3339_MICROS)
```
Comment (dhermes) [inline gcloud/storage/blob.py]: @tseaver This is the part I was referring to. We need to add the timezone here.

**Comment by tseaver** [comment]
This PR fixes the `storage` code which used `isoformat` and `_GOOGLE_TIMESTAMP_FORMAT`.

File: gcloud/storage/blob.py (hunk 1 above)
Comment (tseaver) [inline gcloud/storage/blob.py]: You are correct -- those are naive datetimes. :(

Comment (tseaver) [inline gcloud/storage/blob.py]: 1d35fe8

**Comment by dhermes** [comment]
Everything LGTM if it LGTY

**Pull request merged**